zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java

```diff
@@ -105,74 +106,76 @@
-
-            bindAsContract(InterpreterFactory.class).in(Singleton.class);
-            bindAsContract(NotebookRepoSync.class).to(NotebookRepo.class).in(Singleton.class);
-            bind(LuceneSearch.class).to(SearchService.class).in(Singleton.class);
-            bindAsContract(Helium.class).in(Singleton.class);
-            bind(conf).to(ZeppelinConfiguration.class);
-            bindAsContract(InterpreterSettingManager.class).in(Singleton.class);
-            bindAsContract(InterpreterService.class).in(Singleton.class);
-            bind(credentials).to(Credentials.class);
-            bindAsContract(GsonProvider.class).in(Singleton.class);
-            bindAsContract(WebApplicationExceptionMapper.class).in(Singleton.class);
-            bindAsContract(AdminService.class).in(Singleton.class);
-            bindAsContract(AuthorizationService.class).to(Singleton.class);
-            // TODO(jl): Will make it more beautiful
-            if (!StringUtils.isBlank(conf.getShiroPath())) {
-              bind(ShiroAuthenticationService.class).to(AuthenticationService.class).in(Singleton.class);
-            } else {
-              // TODO(jl): Will be added more type
-              bind(NoAuthenticationService.class).to(AuthenticationService.class).in(Singleton.class);
-            }
-            bindAsContract(HeliumBundleFactory.class).in(Singleton.class);
-            bindAsContract(HeliumApplicationFactory.class).in(Singleton.class);
-            bindAsContract(ConfigurationService.class).in(Singleton.class);
-            bindAsContract(NotebookService.class).in(Singleton.class);
-            bindAsContract(JobManagerService.class).in(Singleton.class);
-            bindAsContract(Notebook.class).in(Singleton.class);
-            bindAsContract(NotebookServer.class)
-                .to(AngularObjectRegistryListener.class)
-                .to(RemoteInterpreterProcessListener.class)
-                .to(ApplicationEventListener.class)
-                .to(NoteEventListener.class)
-                .to(WebSocketServlet.class)
-                .in(Singleton.class);
-            if (conf.isZeppelinNotebookCronEnable()) {
-              bind(QuartzSchedulerService.class).to(SchedulerService.class).in(Singleton.class);
-            } else {
-              bind(NoSchedulerService.class).to(SchedulerService.class).in(Singleton.class);
-            }
-          }
-        });
+            sharedServiceLocator,
+            new AbstractBinder() {
+              @Override
+              protected void configure() {
+                Credentials credentials =
+                        new Credentials(
+                                conf.credentialsPersist(),
+                                conf.getCredentialsPath(),
+                                conf.getCredentialsEncryptKey());
+                bindAsContract(InterpreterFactory.class).in(Singleton.class);
+                bind(conf).to(ZeppelinConfiguration.class);
```

Comment:
Any specific reason to change the order of ZeppelinConfiguration/NotebookRepoSync?